src/rgw/rgw_compression.cc

```diff
@@ -141,8 +141,8 @@ int RGWGetObj_Decompress::handle_data(bufferlist& bl, off_t bl_ofs, off_t bl_len
       return cr;
     }
     ++first_block;
-    while (out_bl.length() - q_ofs >= cct->_conf->rgw_max_chunk_size)
-    {
+    while (out_bl.length() - q_ofs >=
+	   static_cast<off_t>(cct->_conf->rgw_max_chunk_size)) {
```

Comment:
why are the data types changing? it feels like that adds unnecessary risk to this pr